src/interaction/events.jl

```diff
@@ -12,19 +11,65 @@ dropped_files(scene, native_window) = not_implemented_for(native_window)
 hasfocus(scene, native_window) = not_implemented_for(native_window)
 entered_window(scene, native_window) = not_implemented_for(native_window)
 
-function register_callbacks(scene::Scene, native_window)
+function connect_screen(scene::Scene, screen)
+    current_screen = Makie.getscreen(scene)
+    # Already on same screen, nothing to register!
+    current_screen === screen && return
+
+    # different screen, doesn't work yet, since then it would be registered to multiple events
+    if current_screen isa AbstractScreen
+        disconnect_screen(scene, current_screen)
+    end
+
+    window_area(scene, screen)
+    window_open(scene, screen)
+    mouse_buttons(scene, screen)
+    mouse_position(scene, screen)
+    scroll(scene, screen)
+    keyboard_buttons(scene, screen)
+    unicode_input(scene, screen)
+    dropped_files(scene, screen)
+    hasfocus(scene, screen)
+    entered_window(scene, screen)
+
+end
+
+to_native(window::AbstractScreen) = error("to_native(window) not implemented for $(typeof(window)).")
+disconnect!(window::AbstractScreen, signal) = disconnect!(to_native(window), signal)
+
+function disconnect_screen(scene::Scene, screen)
+    delete_screen!(scene, screen)
+    e = events(scene)
+
+    disconnect!(screen, window_area)
+    # disconnect!(e.window_area)
 
-    window_area(scene, native_window)
-    window_open(scene, native_window)
-    mouse_buttons(scene, native_window)
-    mouse_position(scene, native_window)
-    scroll(scene, native_window)
-    keyboard_buttons(scene, native_window)
-    unicode_input(scene, native_window)
-    dropped_files(scene, native_window)
-    hasfocus(scene, native_window)
-    entered_window(scene, native_window)
+    disconnect!(screen, window_open)
+    # disconnect!(e.window_open)
 
+    disconnect!(screen, mouse_buttons)
+    # disconnect!(e.mouse_buttons)
+
+    disconnect!(screen, mouse_position)
+    # disconnect!(e.mouse_position)
+
+    disconnect!(screen, scroll)
+    # disconnect!(e.scroll)
+
+    disconnect!(screen, keyboard_buttons)
+    # disconnect!(e.keyboard_buttons)
+
+    disconnect!(screen, unicode_input)
+    # disconnect!(e.unicode_input)
+
+    disconnect!(screen, dropped_files)
+    # disconnect!(e.dropped_files)
+
+    disconnect!(screen, hasfocus)
+    # disconnect!(e.hasfocus)
+
+    disconnect!(screen, entered_window)
+    # disconnect!(e.entered_window)
 end
```

Comment:
@SimonDanisch What were the `disconnect!(event_obs)` methods supposed to do?

Reply:
Hm, good question. I think I planned to remove any listener, but `connect!` shouldn't really register listeners when I think about it...